### ISSUE TITLE
Add random_state parameter to ClusterConfig

### ISSUE THREAD

**Issue by FBumann** [FBumann/tsam#31]
## Summary

Add a `random_state` parameter to `ClusterConfig` to control reproducibility of non-deterministic clustering methods.

## Problem

Currently, `kmeans` (sklearn KMeans with k-means++ init) and `kmaxoids` (uses `np.random.permutation()`) are non-deterministic. Users must set `np.random.seed()` globally, which is fragile and not discoverable.

## Proposed API

```python
tsam.aggregate(
    data,
    n_clusters=8,
    cluster_config=ClusterConfig(random_state=42),
)
```

## Implementation

- Add `random_state: int | None = None` to `ClusterConfig`
- Pass through to `sklearn.KMeans(random_state=...)` in `period_aggregation.py` / `pipeline/clustering.py`
- Pass through to `k_maxoids.py` (replace bare `np.random.permutation` with a `np.random.RandomState` instance)
- Document which methods are deterministic vs not

## Related

Upstream #103, #101, #129

**Comment by FBumann** [comment]
## Implementation plan

### Code changes (straightforward)

1. **`ClusterConfig`** in `config.py` — add `random_state: int | None = None` field
2. **`period_aggregation.py`** — thread `random_state` through `aggregate_periods()`, pass to `KMeans(random_state=...)` and `KMaxoids(random_state=...)`
3. **`pipeline/clustering.py`** — pass `cluster.random_state` in both `cluster_periods()` and `cluster_sorted_periods()`
4. **`utils/k_maxoids.py`** — replace global `numpy.random.permutation()` with `np.random.RandomState(self.random_state)` instance

### Test regeneration (the heavy part)

Changing from global `np.random.seed()` to per-estimator `random_state` changes the RNG path for kmaxoids (and potentially kmeans). This means:

- **`test/data/clustering_e2e/`** — kmaxoids (and possibly kmeans) fixture CSVs need regeneration via `generate_fixtures()`
- **`test/data/golden/`** — kmaxoids golden files need regeneration via `pytest --update-golden`
- **`test_old_new_equivalence.py`** — needs to pass `random_state=SEED` via `ClusterConfig` instead of calling `np.random.seed()` globally
- **`test_clustering_e2e.py`** — same: pass `random_state=SEED` via `ClusterConfig`
- The fixture regeneration also picks up the v4 column-order change (input order instead of alphabetical), causing large diffs in all fixture files even for deterministic methods

### Verified working

Smoke test confirms `random_state` works correctly:
```python
r1 = tsam.aggregate(data, n_clusters=4, cluster=ClusterConfig(method='kmeans', random_state=42))
r2 = tsam.aggregate(data, n_clusters=4, cluster=ClusterConfig(method='kmeans', random_state=42))
assert r1.cluster_representatives.equals(r2.cluster_representatives)  # True

r3 = tsam.aggregate(data, n_clusters=4, cluster=ClusterConfig(method='kmeans', random_state=99))
assert not r1.cluster_representatives.equals(r3.cluster_representatives)  # True
```

### Decision

Deferring this to a dedicated PR since the fixture regeneration touches many files and should be reviewed carefully alongside the RNG behavior change.
